token-metadata/program/src/utils/master_edition.rs

```diff
@@ -96,49 +98,105 @@ pub fn process_mint_new_edition_from_master_edition_via_token_logic<'a>(
         return Err(MetadataError::AlreadyInitialized.into());
     }
 
-    let edition_number = edition.checked_div(EDITION_MARKER_BIT_SIZE).unwrap();
-    let as_string = edition_number.to_string();
-
-    let bump = assert_derivation(
-        program_id,
-        edition_marker_info,
-        &[
-            PREFIX.as_bytes(),
-            program_id.as_ref(),
-            master_metadata.mint.as_ref(),
-            EDITION.as_bytes(),
-            as_string.as_bytes(),
-        ],
-    )?;
-
-    if edition_marker_info.data_is_empty() {
-        let seeds = &[
-            PREFIX.as_bytes(),
-            program_id.as_ref(),
-            master_metadata.mint.as_ref(),
-            EDITION.as_bytes(),
-            as_string.as_bytes(),
-            &[bump],
-        ];
+    match token_standard {
+        TokenStandard::NonFungibleEdition => {
+            let edition_number = edition.checked_div(EDITION_MARKER_BIT_SIZE).unwrap();
+            let as_string = edition_number.to_string();
+
+            let bump = assert_derivation(
+                program_id,
+                edition_marker_info,
+                &[
+                    PREFIX.as_bytes(),
+                    program_id.as_ref(),
+                    master_metadata.mint.as_ref(),
+                    EDITION.as_bytes(),
+                    as_string.as_bytes(),
+                ],
+            )?;
+
+            if edition_marker_info.data_is_empty() {
+                let seeds = &[
+                    PREFIX.as_bytes(),
+                    program_id.as_ref(),
+                    master_metadata.mint.as_ref(),
+                    EDITION.as_bytes(),
+                    as_string.as_bytes(),
+                    &[bump],
+                ];
+
+                create_or_allocate_account_raw(
+                    *program_id,
+                    edition_marker_info,
+                    system_account_info,
+                    payer_account_info,
+                    MAX_EDITION_MARKER_SIZE,
+                    seeds,
+                )?;
+            }
 
-        create_or_allocate_account_raw(
-            *program_id,
-            edition_marker_info,
-            system_account_info,
-            payer_account_info,
-            MAX_EDITION_MARKER_SIZE,
-            seeds,
-        )?;
-    }
+            let mut edition_marker = EditionMarker::from_account_info(edition_marker_info)?;
+            edition_marker.key = Key::EditionMarker;
+            if edition_marker.edition_taken(edition)? {
+                return Err(MetadataError::AlreadyInitialized.into());
+            } else {
+                edition_marker.insert_edition(edition)?
+            }
+            edition_marker.serialize(&mut *edition_marker_info.data.borrow_mut())?;
+            Ok::<(), ProgramError>(())
```

Comment:
It required it for at least one of the Oks. I added it to both for consistency.

Reply:
Its only needed because `?` is on the match statement, so it needs the fully qualified types.  If that is removed then you don't need to have any `Ok` here.